src/pages/ThorChainLP/hooks/useUserLpData.ts

```diff
@@ -0,0 +1,106 @@
+import { type AccountId, type AssetId, thorchainAssetId } from '@shapeshiftoss/caip'
+import { useQuery } from '@tanstack/react-query'
+import axios from 'axios'
+import { getConfig } from 'config'
+import { useMemo } from 'react'
+import { bn } from 'lib/bignumber/bignumber'
+import type { ThornodePoolResponse } from 'lib/swapper/swappers/ThorchainSwapper/types'
+import { assetIdToPoolAssetId } from 'lib/swapper/swappers/ThorchainSwapper/utils/poolAssetHelpers/poolAssetHelpers'
+import { fromThorBaseUnit } from 'lib/utils/thorchain'
+import { getThorchainLendingPosition } from 'lib/utils/thorchain/lending'
+import { getThorchainLiquidityProviderPosition } from 'lib/utils/thorchain/lp'
+import { selectMarketDataById } from 'state/slices/marketDataSlice/selectors'
+import { useAppSelector } from 'state/store'
+
+type UseLendingPositionDataProps = {
+  accountId: AccountId | undefined
+  assetId: AssetId
+}
+
+const calculatePoolOwnershipPercentage = (
+  userLiquidityUnits: string,
+  totalPoolUnits: string,
+): string => bn(userLiquidityUnits).div(totalPoolUnits).times(100).toFixed()
+
+export const thorchainLendingPositionQueryFn = async ({
+  queryKey,
+}: {
+  queryKey: [string, { accountId: AccountId; assetId: AssetId }]
+}) => {
+  const [, { accountId, assetId }] = queryKey
+  const position = await getThorchainLendingPosition({ accountId, assetId })
+  return position
+}
+
+export const useUserLpData = ({ accountId, assetId }: UseLendingPositionDataProps) => {
+  // TODO(gomes): return a list of positions, and discriminate asym/sim as two different positions
```

Comment:
Tackle me before opening - we absolutely want to handle this as early as possible

Reply:
Actually not going to be tackled just yet but in a follow-up - getting the asym/sym right is quite complex, better to start with one happy case (asym on the asset side) and iterate from there